src/RuleCommonITILObject.php

```diff
@@ -371,61 +371,105 @@ public function executeActions($output, $params, array $input = [])
                         break;
 
                     case 'regex_result':
-                        if ($action->fields["field"] == "_affect_itilcategory_by_code") {
-                            if (isset($this->regex_results[0])) {
-                                $regexvalue = RuleAction::getRegexResultById(
-                                    $action->fields["value"],
-                                    $this->regex_results[0]
-                                );
-                            } else {
-                                $regexvalue = $action->fields["value"];
-                            }
+                        // Get each regex values
+                        $regex_values = array_map(
+                            fn ($regex_result) => RuleAction::getRegexResultById(
+                                $action->fields["value"],
+                                $regex_result
+                            ),
+                            $this->regex_results
+                        );
 
-                            if (!is_null($regexvalue)) {
-                                $target_itilcategory = ITILCategory::getITILCategoryIDByCode($regexvalue);
+                        // Get field
+                        $field = $action->fields["field"];
+
+                        // Handle each fields
+                        if ($field == "_affect_itilcategory_by_code") {
```

Comment:
You could probably use a switch here, instead of if/elseif/elseif/....

Reply:
I'd like to avoid a switch in a switch for readability, and I think there is already too much nesting...  
IMO the regex_result case should have it's own dedicated function.